lib/OnyxUtils.ts

```diff
@@ -629,7 +666,16 @@ function keyChanged<TKey extends OnyxKey>(
     // We are iterating over all subscribers to see if they are interested in the key that has just changed. If the subscriber's  key is a collection key then we will
     // notify them if the key that changed is a collection member. Or if it is a regular key notify them when there is an exact match. Depending on whether the subscriber
     // was connected via withOnyx we will call setState() directly on the withOnyx instance. If it is a regular connection we will pass the data to the provided callback.
-    const stateMappingKeys = Object.keys(callbackToStateMapping);
+    let stateMappingKeys = onyxKeyToConnectionIDs.get(key);
+
+    if (!stateMappingKeys) {
+        // Getting the collection key from the specific key 'cos only collection keys were stored in the mapping.
+        stateMappingKeys = onyxKeyToConnectionIDs.get(getPureKey(key));
+        if (!stateMappingKeys) {
```

Comment:
Could you update the comment above to explain this thoroughly? Also can you explain here and in comment why do we not need the same change for the `keysChanged` method?